pkg/vault/test/foundry/BatchRouterNestedPools.t.sol

```diff
@@ -0,0 +1,351 @@
+            MAX_ROUND_ERROR,
+            "ChildPoolB Dai Balance is wrong"
+        );
+        assertEq(
+            vars.childPoolBAfter.wsteth,
+            vars.childPoolBBefore.wsteth - amountsOut[vars.wstethIdx],
+            "ChildPoolB Wsteth Balance is wrong"
+        );
+
+        // Check ParentPool
+        assertApproxEqAbs(
+            vars.parentPoolAfter.dai,
+            vars.parentPoolBefore.dai -
+                (amountsOut[vars.daiIdx] - (poolInitAmount - (MIN_BPT / 2)).mulDown(proportionToRemove)),
+            MAX_ROUND_ERROR,
+            "ParentPool Dai Balance is wrong"
+        );
+        assertEq(
+            vars.parentPoolAfter.childPoolABpt,
+            vars.parentPoolBefore.childPoolABpt - burnedChildPoolABpts,
+            "ParentPool ChildPoolA BPT Balance is wrong"
+        );
+        assertEq(
+            vars.parentPoolAfter.childPoolBBpt,
+            vars.parentPoolBefore.childPoolBBpt - burnedChildPoolBBpts,
+            "ParentPool ChildPoolB BPT Balance is wrong"
+        );
+    }
+
+    function testRemoveLiquidityNestedPoolLimits() public {
+        // Remove 10% of pool liquidities.
+        uint256 proportionToRemove = 10e16;
+
+        uint256 totalPoolBPTs = BalancerPoolToken(parentPool).totalSupply();
+        // Since LP is the owner of all BPT supply, and part of the BPTs were burned in the initialization step, using
+        // totalSupply is more accurate to remove exactly the proportion that we intend from each pool.
+        uint256 exactBptIn = totalPoolBPTs.mulDown(proportionToRemove);
+
+        NestedPoolTestLocals memory vars = _createNestedPoolTestLocals();
+
+        // During pool initialization, MIN_BPT amount of BPT is burned to address(0), so that the pool cannot be
+        // completely drained. We need to discount this amount of tokens from the total liquidity that we can extract
+        // from the child pools.
+        uint256 deadTokens = (MIN_BPT / 2).mulDown(proportionToRemove);
+
+        uint256[] memory minAmountsOut = new uint256[](4);
+        // Expect minAmountsOut to be the liquidity of the pool, which is more than what should return.
+        minAmountsOut[vars.daiIdx] = poolInitAmount;
+        minAmountsOut[vars.wethIdx] = poolInitAmount;
+        minAmountsOut[vars.wstethIdx] = poolInitAmount;
+        minAmountsOut[vars.usdcIdx] = poolInitAmount;
+
+        // DAI exists in childPoolB and parentPool, so we expect 2x more DAI than the other tokens.
+        // Since pools are in their initial state, we can use poolInitAmount as the balance of each token in the pool.
+        // Also, we only need to account deadTokens once, since we calculate the bpts in for the parent pool using
+        // totalSupply (so the burned MIN_BPT amount does not affect the bpt in calculation and the amounts out are
+        // perfectly proportional to the parent pool balance)
```

Comment:
```suggestion
        // Also, we only need to account for deadTokens once, since we calculate the BPT in for the parent pool using
        // totalSupply (so the burned MIN_BPT amount does not affect the BPT in circulation, and the amounts out are
        // perfectly proportional to the parent pool balance).

```